### PR TITLE
Rewrite client config api

### PR DESCRIPTION
**Pull request by runeflobakk**
### 💰 Description

Redesign client configuration API.

### 🏆 Highlights

- redesign client timeouts configuration
- Removed method in `ClientConfiguration` to enable request and response logging. This method was previously necessary with Jersey Client, but equivalent logging with Apache Http Client is enabled just by configuring certain loggers. https://hc.apache.org/httpcomponents-client-5.2.x/logging.html
- The concept of a _default sender_ (previously "global sender") for the client instance, which is used if not specifying the sender per job, has been captured using the (internal) type `MaySpecifySender`, which has support for the fallback scenario. This is instead of passing around `Optional<Sender>` variables, which had to be resolved by (now deleted) `ActualSender.getActualSender(..)`

### 👀 Examples 

TODO: some API examples

### ⚙️ Dependencies

- [ ] Updating documentation on logging requests and responses: https://signering-docs.readthedocs.io/en/latest/client-integration/debugging.html#java
- [ ] Update documentation for ClientConfiguration: https://signering-docs.readthedocs.io/en/latest/client-integration/create-client-configuration.html#client-configuration

### CODE REVIEW

**Comment by runeflobakk** [comment]
Denne begynner egentlig å bli klar for review. Legger til alle utviklere på teamet, så kan dere som vil det delta i ev. diskusjon!
Skal få lagt til litt eksempler på nytt API, som følgelig også må inn i dokumentasjon når vi lander endelig release.

File: src/main/java/no/digipost/signature/client/ClientConfiguration.java [hunk 1]
```diff
@@ -143,144 +124,153 @@ public static Builder builder(KeyStoreConfig keystore) {
...
+                    .socketTimeout(Duration.ofSeconds(5))
+                    .connectTimeout(Duration.ofSeconds(5))
+                    .connectionRequestTimeout(Duration.ofSeconds(5))
+                    .responseArrivalTimeout(Duration.ofSeconds(10));
+            this.httpClientForDocumentDownloadsConfigurer = new ApacheHttpClientBuilderConfigurer()
+                    .connectionManager(sslConfigurer)
+                    .socketTimeout(Duration.ofSeconds(5))
+                    .connectTimeout(Duration.ofSeconds(10))
+                    .connectionRequestTimeout(Duration.ofSeconds(10))
+                    .responseArrivalTimeout(Duration.ofSeconds(60));
```
Comment (runeflobakk) [inline src/main/java/no/digipost/signature/client/ClientConfiguration.java:158]: Foreløpig forslag til timeout defaults.

File: src/test/java/no/digipost/signature/client/core/internal/security/TrustStoreLoaderTest.java [hunk 2]
```diff
@@ -69,8 +49,7 @@ void loads_productions_certificates() throws KeyStoreException {
 
     @Test
     void loads_test_certificates() throws KeyStoreException {
-        ClientConfiguration config = configBuilder.trustStore(TEST).build();
-        KeyStore trustStore = TrustStoreLoader.build(config);
+        KeyStore trustStore = TrustStoreLoader.build(DIFITEST);
```
Comment (simenstoa) [inline src/test/java/no/digipost/signature/client/core/internal/security/TrustStoreLoaderTest.java:52]: Skal vi bytte til `STAGING` her også?

File: src/test/java/no/digipost/signature/client/docs/InitialSetup.java [hunk 3]
```diff
@@ -39,9 +39,8 @@ static void create_client_configuration() {
         KeyStoreConfig keyStoreConfig = null; //As initialized earlier
 
         ClientConfiguration clientConfiguration = ClientConfiguration.builder(keyStoreConfig)
-                .trustStore(Certificates.TEST)
-                .serviceUri(ServiceUri.DIFI_TEST)
-                .globalSender(new Sender("123456789"))
+                .serviceEnvironment(DIFITEST)
```
Comment (simenstoa) [inline src/test/java/no/digipost/signature/client/docs/InitialSetup.java:42]: Og her?

File: src/test/java/no/digipost/signature/client/core/internal/security/TrustStoreLoaderTest.java (hunk 2 above)
Comment (runeflobakk) [reply src/test/java/no/digipost/signature/client/core/internal/security/TrustStoreLoaderTest.java:52]: Jepp

**Review by simenstoa — APPROVED** [review]
👏

**Pull request merged**